proto/internal/temporal/server/api/schedule/v1/message.proto

```diff
@@ -155,3 +159,55 @@ message HsmSchedulerState {
     google.protobuf.Timestamp next_invocation_time = 3;
 
 }
+
+// State machine scheduler internal state.
+message SchedulerInternal {
+    temporal.server.api.enums.v1.Scheduler2State state = 1;
+
+    // Scheduler request parameters and metadata. 
+    temporal.api.schedule.v1.Schedule schedule = 2;
+    temporal.api.schedule.v1.ScheduleInfo info = 3;
+    temporal.api.schedule.v1.SchedulePatch initial_patch = 4;
+
+    // State common to all generators is stored in the top-level machine.
+    string namespace = 5;
+    string namespace_id = 6;
+    string schedule_id = 7;
+
+    // Implemented as a sequence number. Used for optimistic locking against
+    // update requests.
+    int64 conflict_token = 8;
+}
+
+// State machine scheduler's Generator internal state.
+message GeneratorInternal {
+    temporal.server.api.enums.v1.SchedulerGeneratorState state = 1;
+
+    // Generator waits for the next_invocation_time before buffering more actions.
+    google.protobuf.Timestamp next_invocation_time = 2;
+
+    // High water mark.
+    google.protobuf.Timestamp last_processed_time = 3;
+}
+
+// State machine scheduler's Executor internal state.
+message ExecutorInternal {
+    temporal.server.api.enums.v1.SchedulerExecutorState state = 1;
+
+    // Wakes immediately after actions are buffered, or when the deadline from
+    // the BACKING_OFF state has expired.
+    google.protobuf.Timestamp next_invocation_time = 2;
+
+    // Buffered starts that will be started by the Executor.
+    repeated BufferedStart buffered_starts = 3;
+}
+
+// State machine scheduler's Backfiller internal state.
+message BackfillerInternal {
```

Comment:
I was imagining that we'd have multiple backfillers and each would have their own backfill config attached.